src/Compilers/Extension/CompilerPackage.cs

```diff
@@ -152,7 +152,18 @@ private async Task<string> GetMSBuildVersionStringAsync(CancellationToken cancel
                 throw new Exception($"Unrecognized Visual Studio Version: {dte.Version}");
             }
 
-            return parts[0] + ".0";
+            int majorVersion = int.Parse(parts[0]);
+
+            if (majorVersion >= 16)
```

Comment:
💭 

```csharp
if (int.TryParse(parts[0], out var majorVersion) && majorVersion >= 16)
```

Reply:
If that's not a number...I want it crashing. :-)